jerry-ext/handler/handler-register.c

```diff
@@ -40,3 +40,148 @@ jerryx_handler_register_global (const jerry_char_t *name_p, /**< name of the fun
+ *         In case of error the (error object, registered property count) pair.
+ */
+jerryx_register_result
+jerryx_set_properties (const jerry_value_t target_object, /**< target object */
+                       const jerryx_property_entry entries[]) /**< array of method entries */
+{
+  uint32_t idx = 0;
+  for (; ((entries + idx) != NULL) && (entries[idx].name != NULL); idx++)
+  {
+    const jerryx_property_entry* entry = &entries[idx];
+
+    jerry_value_t prop_name = jerry_create_string_from_utf8 ((const jerry_char_t *) entry->name);
+    jerry_value_t result = jerry_set_property (target_object, prop_name, entry->value);
+
+    jerry_release_value (prop_name);
+
+    // By API definition:
+    // The jerry_set_property returns TRUE if there is no problem
+    // and error object if there is any problem.
+    // Thus there is no need to check if the boolean value is false or not.
+    if (!jerry_value_is_boolean (result)) {
+      return (jerryx_register_result) { result, idx };
+    }
+
+    jerry_release_value (entry->value);
+    jerry_release_value (result);
+  }
+
+  return (jerryx_register_result) { jerry_create_undefined (), idx };
+
+} /* jerryx_set_properties */
+
+
+/**
+ * Release all jerry_value_t in a jerryx_property_entry array based on
+ * a previous jerryx_set_properties call.
+ *
+ * In case of a successful registration it is safe to call this method.
+ */
+void
+jerryx_free_property_entry (const jerryx_property_entry entries[], /**< list of property entries */
+                            const jerryx_register_result register_result) /**< previous result of registration */
+{
+  for (uint32_t idx = register_result.registered;
+       ((entries + idx) != NULL) && (entries[idx].name != NULL);
+       idx++)
+  {
+    jerry_release_value (entries[idx].value);
+  }
+} /* jerryx_property_entry_cleanup */
+
+/**
+ * Set a property to a specified value with a given name.
+ *
+ * Notes:
+ *   - The operation performed is the same as what the `jerry_set_property` method.
+ *   - The property name must be a zero terminated UTF-8 string.
+ *   - There should be no '\0' (NULL) character in the name excluding the string terminator.
+ *   - Returned value must be freed with jerry_release_value, when it is no longer needed.
+ *
+ * @return true value - if the operation was successful
+ *         thrown error - otherwise
+ */
+jerry_value_t
+jerryx_set_property_str (const jerry_value_t target_object, /**< target object */
+                         const char *name, /**< property name */
+                         const jerry_value_t value) /**< value to set */
+{
+  jerry_value_t property_name_val = jerry_create_string_from_utf8 ((const jerry_char_t *) name);
+  jerry_value_t result_val = jerry_set_property (target_object, property_name_val, value);
+
+  jerry_release_value (property_name_val);
+
+  return result_val;
+} /* jerryx_set_property_str */
+
+
```

Comment:
Please fix the double newlines inbetween the added functions, there are several occurrences.